### PR TITLE
[Bug fix] port_alias.py enhancement for an index condition judge

### PR DESCRIPTION
**Pull request by keboliu**
### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [x] Bug fix
- [] Testbed and Framework(new/improvement)
- [] Test case(new/improvement)

#### How did you do it?
Add an index condition check to make sure the index will not out of range.

#### How did you verify/test it?
run generate minigraph command

#### Any platform specific information?
No

#### Supported testbed topology if it's a new test case?
support all topo

### CODE REVIEW

File: ansible/library/port_alias.py [hunk 1]
```diff
@@ -112,7 +112,7 @@ def get_portmap(self):
                     aliases.append(alias)
                     portmap[name] = alias
                     aliasmap[alias] = name
-                    if speed_index != -1:
+                    if (speed_index != -1) and (len(mapping) > speed_index):
```
Comment (nazariig) [inline ansible/library/port_alias.py]:
```python
alias_index = -1
speed_index = -1
title = None
while len(lines) != 0:
...
if re.match('^\w', line):
    mapping = line.split()
    if title and len(mapping) == len(title):
        ...
        if speed_index != -1:
            portspeed[alias] = mapping[speed_index]
```

Comment (keboliu) [reply ansible/library/port_alias.py]: In the code to parse port_config.ini, seems that even len(mapping) != len(title), in some cases the port_config.ini could still be valid.  https://github.com/Azure/sonic-platform-common/blob/master/sonic_sfp/sfputilbase.py#L400

Comment (qiluo-msft) [inline ansible/library/port_alias.py]: >(speed_index != -1) and (len(mapping) > speed_index) [](start = 23, length = 52)

speed_index in mapping

Comment (qiluo-msft) [inline ansible/library/port_alias.py]: >if [](start = 20, length = 2)

Otherwise, should we throw?

Comment (keboliu) [reply ansible/library/port_alias.py]: speed_index is just an index, not a string, so it couldn't be a member of mapping. we can not use membership operator here.

Comment (keboliu) [reply ansible/library/port_alias.py]: I think we shouldn't throw here, if the last column of the port_config.ini is "speed", then it's OK to have some ports left w/o speed explicitly specified(at least in our practice we have this case).  In this case,  means the speed on this port doesn't matter, in the later on minigraph generate process, will assign a default speed value to it.

**Pull request merged**